### PR TITLE
Use expectAssertion for absorbing dev vs prod differences.

### PR DESCRIPTION
**Pull request by rwjblue**

### CODE REVIEW

File: packages/ember-runtime/tests/system/object/create_test.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ QUnit.test('inherits properties from passed in EmberObject', function() {
 QUnit.test('throws if you try to pass anything a string as a parameter', function() {
   let expected = 'EmberObject.create only accepts an objects.';
 
-  throws(() => EmberObject.create('some-string'), expected);
+  expectAssertion(() => EmberObject.create('some-string'), expected);
```
Comment (mixonic) [inline packages/ember-runtime/tests/system/object/create_test.js]: Confirmed this is an assert. The title of the test is incorrect.

File: packages/ember/tests/helpers/link_to_test/link_to_with_query_params_test.js [hunk 2]
```diff
@@ -564,7 +564,7 @@ moduleFor('The {{link-to}} helper + query params - globals mode app', class exte
   [`@test the {{link-to}} helper throws a useful error if you invoke it wrong`](assert) {
     assert.expect(1);
 
-    assert.throws(() => {
+    expectAssertion(() => {
```
Comment (mixonic) [inline packages/ember/tests/helpers/link_to_test/link_to_with_query_params_test.js]: This is also an assertion.

**Review by mixonic — APPROVED** [review]
LGTM.

**Comment by mixonic** [comment]
Kicked the build here.

**Comment by mmun** [comment]
This was merged in https://github.com/emberjs/ember.js/commit/db55502dee8d6eb31f31d88bf407ccc17b7d1e9c.

**Pull request closed without merging**